clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3799,7 +3784,8 @@ void Sema::CheckFunctionBodyBoundsDecls(FunctionDecl *FD, Stmt *Body) {
     // __finally or may encounter a malformed AST.  Fall back on to non-flow 
     // based analysis.  The CSS parameter is ignored because the checked
     // scope information is obtained from Body, which is a compound statement.
-    Checker.TraverseStmt(Body, CheckedScopeSpecifier::CSS_Unchecked);
+    Checker.SetCheckedScopeSpecifier(CheckedScopeSpecifier::CSS_Unchecked);
```

Comment:
I don't believe this is the proper use of the RAII ("resource allocation is acquisition") idiom.  I think you want to create an object that on construction stores the existing CSS value and sets the new CSS value, and on destruction restores the old CSS value.